.github/workflows/pythonpublish.yml

```
@@ -16,6 +16,12 @@ jobs:
     - name: Set up Python
       run: uv python install 3.11
 
+    - name: Create the virtual environment
+      run: |
+        # Change directory so that xdsl-opt can be found during installation.
+        cd xdsl
+        VENV_EXTRAS="--all-extras" make venv
+
```

```suggestion
    - name: Build and publish
      run: |
        uvx --with setuptools --with wheel setup.py sdist bdist_wheel
        uvx twine upload -u __token__ -p ${{ secrets.PYPI_XDSL_TOKEN }} dist/*
```
I had a quick read of the docs here and it looks like the lines below can be done like this, the point is to not use the same venv as the xdsl one for these one-off scripts. BTW, do we want to reuse the uv cache for this?

https://docs.astral.sh/uv/concepts/tools/#the-path

---

I wouldn't use the cache here, we run this once every few weeks or so, so we don't need it?